src/pip/_internal/network/auth.py

```diff
@@ -219,8 +219,9 @@ def _prompt_for_password(self, netloc):
         if not username:
             return None, None
         auth = get_keyring_auth(netloc, username)
-        if auth:
-            return auth[0], auth[1], False
+        if auth is not None:
+            if auth[0] is not None and auth[1] is not None:
+                return auth[0], auth[1], False
```

Comment:
```suggestion
        if auth is not None and auth[0] is not None and auth[1] is not None:
            return auth[0], auth[1], False
```

Reply:
I used nested one to keep the code cleaner. I though keeping all conditions was too much in a single line.